pkg/goDB/conditions/node/filter_direction_test.go

```diff
@@ -0,0 +1,132 @@
+package node
+
+import (
+	"errors"
+	"fmt"
+	"github.com/els0r/goProbe/pkg/types"
+	"github.com/els0r/goProbe/pkg/types/hashmap"
+	"reflect"
+	"testing"
+)
+
+var isDirectionConditionTest = []struct {
+	node        conditionNode
+	expectedErr error
+}{
+	{node: conditionNode{attribute: "dir", comparator: "=", value: "in"}, expectedErr: errMisplacedDirectionFilterCondition},
+	{node: conditionNode{attribute: "sip", comparator: "=", value: "127.0.0.1"}, expectedErr: nil},
+}
+
+func TestIsDirectionCondition(t *testing.T) {
+	for _, test := range isDirectionConditionTest {
+		n, err := isDirectionCondition(test.node)
+		if !errors.Is(err, test.expectedErr) {
```

Comment:
Maybe make your life easier by using the functionality of `testify` (we're relying on it all around the place) instead of manually handling errors + `Fatal()` etc...